examples/fixture/asm/r1cs/public_output_array.asm

```diff
@@ -0,0 +1,6 @@
+@ noname.0.7.0
+
+2 == (v_3 + v_4) * (1)
+v_5 == (v_3 + v_4) * (v_3)
+v_3 + v_4 + 6 == (v_1) * (1)
+v_5 == (v_2) * (1)
```

Comment:
^ this last constraint seems really useless, we should optimize that, I'll create an issue :)

Reply:
Do you mean the second constraint should be instead 
`v_2 = (v_3 + v_4) * (v_3)`?

`v_1` and `v_2` are the outputs and they have to be constrained

Reply:
yeah! `v_2 == (v_3 + v_4) * v_3` would have been better. But I don't think we should look to create such optimization, because an optimization pass that would do inlining would easily inline the last constraint